### PR TITLE
Allow Perforce change sources to resolve the 'who' field into a more verbose format using a callable

### PR DESCRIPTION
**Pull request by sphet**
* [X] I have updated the unit tests
* [X] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
* [X] I have updated the appropriate documentation

### CODE REVIEW

File: master/buildbot/changes/p4poller.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def __init__(self, p4port=None, p4user=None, p4passwd=None,
                  pollInterval=60 * 10, histmax=None, pollinterval=-2,
                  encoding='utf8', project=None, name=None,
                  use_tickets=False, ticket_login_interval=60 * 60 * 24,
-                 server_tz=None, pollAtLaunch=False, revlink=lambda branch, revision: (u'')):
+                 server_tz=None, pollAtLaunch=False, revlink=lambda branch, revision: (u''), resolvewho=lambda who: (who)):
```
Comment (seankelly) [inline master/buildbot/changes/p4poller.py]: Any way this could be done as a renderable instead?

Comment (sphet) [reply master/buildbot/changes/p4poller.py]: @seankelly - I'm not sure I understand how a renderable would be appropriate here. I'm open to any suggestions though. 

In our case we take the 'who' in the form of user@workspace - for example Charles@SecondWorkspace - and need to map it to charles.oneal@workspace.com. We pipe it through a callable that has LDAP and perforce user list knowledge, so it is a couple of dictionary checks and some additional logic. How do renderables work in this case?

Comment (seankelly) [reply master/buildbot/changes/p4poller.py]: A renderable can be a function to do that. It meshes better with how Buildbot handles dynamic text. The issue is that its input is the build's properties.

I'm hoping @tardyp might have an idea.

**Comment by sphet** [comment]
Sean

I suspect that because this happens at the poller/changes level it is "too
soon" to be part of the build renderables?

File: master/buildbot/changes/p4poller.py (hunk 1 above)
Comment (tardyp) [reply master/buildbot/changes/p4poller.py]: yeah, I've thought of that too, but renderable really do not have anything to interpolate in pollers

**Comment by sphet** [comment]
@tardyp Can you point me in the direction (an example) of writing a unit test to test the config error? Thanks.

**Comment by tardyp** [comment]
https://github.com/buildbot/buildbot/blob/master/master/buildbot/test/unit/test_changes_gitpoller.py#L1340

File: master/buildbot/test/unit/test_changes_p4poller.py [hunk 2]
```diff
@@ -441,7 +441,9 @@ def check(res):
         d.addCallback(check)
         return d
 
-
+    def test_resolveWho_callable(self):
+        self.assertRaisesConfigError("You need to provide a valid callable for resolvewho",
```
Comment (tardyp) [inline master/buildbot/test/unit/test_changes_p4poller.py]: you need to inherit configmixin to have this check available

Comment (sphet) [reply master/buildbot/test/unit/test_changes_p4poller.py]: Thanks - got sidetracked in the office today and forgot to check my update PR

**Comment by sphet** [comment]
Not sure what those remaining failures are caused by.

**Comment by tardyp** [comment]
Looks like the remaining failures are unusually high false positive rate. Can you please squash your history and force push? This will restart the CI

**Comment by tardyp** [comment]
thanks!

**Pull request merged**